### PR TITLE
Move modules/forms to services/forms

### PR DESCRIPTION
**Pull request by zeripath**
Forms are dependent on models and therefore should be in services.

This PR also removes the old auth. aliasing

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by 6543** [comment]
I'm not sure if service is the right place modules sounds right ...
https://github.com/go-gitea/gitea/blob/master/CONTRIBUTING.md#design-guideline

File: routers/api/v1/api.go [hunk 1]
```diff
@@ -897,7 +897,7 @@ func Routes() *web.Route {
 						m.Get(".patch", repo.DownloadPullPatch)
 						m.Post("/update", reqToken(), repo.UpdatePullRequest)
 						m.Combo("/merge").Get(repo.IsPullRequestMerged).
-							Post(reqToken(), mustNotBeArchived, bind(auth.MergePullRequestForm{}), repo.MergePullRequest)
```
Comment (6543) [inline routers/api/v1/api.go:900]: struct should be moved into /modules/structs - if it is needed for webUI too copied

**Review by lunny — APPROVED** [review]

File: routers/api/v1/api.go (hunk 1 above)
Comment (lunny) [reply routers/api/v1/api.go:900]: I don't think move all structs to structs module is a good idea. In fact, if there is any other chance, I think move structs to other modules is better. But I have no idea now.

**Comment by zeripath** [comment]
@6543 modules are not supposed to use models.

```
services: Support functions for common routing operations. Uses models and modules to handle the request.
```

* they depend on http
* these are web forms.
* the only places that use forms are routers and integrations:

```
find -name *.go | xargs grep modules/form                                                  [15:06:30]
./routers/routes/web.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/routes/install.go:	"code.gitea.io/gitea/modules/forms"
./routers/repo/branch.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/release.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/milestone.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/issue_timetrack.go:	"code.gitea.io/gitea/modules/forms"
./routers/repo/issue.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/projects.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/repo.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/pull_review.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/release_test.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/webhook.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/pull.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/issue_label_test.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/setting.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/wiki_test.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/issue_lock.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/editor.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/migrate.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/issue_label.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/settings_test.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/setting_protected_branch.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/repo/wiki.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/install.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/user/auth.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/user/oauth.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/user/setting/oauth2.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/user/setting/account.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/user/setting/account_test.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/user/setting/applications.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/user/setting/security_twofa.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/user/setting/security_openid.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/user/setting/keys.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/user/setting/profile.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/user/setting/security_u2f.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/user/auth_openid.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/api/v1/repo/pull.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/api/v1/repo/migrate.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/api/v1/api.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/api/v1/swagger/options.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/admin/users_test.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/admin/auths.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/admin/users.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/admin/admin.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/org/org.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/org/setting.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/org/teams.go:	auth "code.gitea.io/gitea/modules/forms"
./routers/org/org_labels.go:	auth "code.gitea.io/gitea/modules/forms"
./integrations/api_helper_for_declarative_test.go:	auth "code.gitea.io/gitea/modules/forms"
./integrations/api_pull_test.go:	auth "code.gitea.io/gitea/modules/forms"
```

* modules are not supposed to use models - this uses models

They definitely fit the service spec.

**Review by 6543 — APPROVED** [review]

**Comment by zeripath** [comment]
make lgtm work

**Pull request merged**